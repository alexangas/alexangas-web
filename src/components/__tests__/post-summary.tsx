import React from "react"
import renderer from "react-test-renderer"

import PostSummary from "../post-summary"

describe(`PostSummary`, () => {
  it(`renders post summary with description`, () => {
    const testRenderer = renderer.create(
      <PostSummary
        slug="posttitle"
        title="Post Title"
        description="The description of the post."
        dateTime="2020-05-08T12:58"
        tags={[`tagA`, `tagB`, `tagC`]}
      />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`renders post summary with excerpt`, () => {
    const testRenderer = renderer.create(
      <PostSummary
        slug="posttitle"
        title="Post Title"
        excerpt="The excerpt of the post."
        dateTime="2020-05-08T12:58"
        tags={[`tagA`, `tagB`, `tagC`]}
      />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
