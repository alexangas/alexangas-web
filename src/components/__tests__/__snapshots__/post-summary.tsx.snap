// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSummary renders post summary with description 1`] = `
<div
  className="content"
>
  <h2
    className="subtitle is-marginless is-4"
  >
    <a
      href="/blogposttitle"
    >
      Post Title
    </a>
  </h2>
  <p
    className="is-marginless"
    dangerouslySetInnerHTML={
      Object {
        "__html": "The description of the post.",
      }
    }
  />
  <div
    className="blog-metadata"
  >
    <div
      className="is-inline-block-tablet"
    >
      <small>
        <time
          dateTime="2020-05-08T12:58"
        >
          8 May 2020
        </time>
      </small>
    </div>
    <span
      className="blog-metadata-separator is-hidden-mobile"
    />
    <div
      className="tags is-inline-block-tablet"
    >
      <a
        className="tag is-paddingless"
        href="/blog/tags/tag-a/"
      >
        <span
          className="icon mr-0"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
            />
          </svg>
        </span>
        tagA
      </a>
      <a
        className="tag is-paddingless"
        href="/blog/tags/tag-b/"
      >
        <span
          className="icon mr-0"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
            />
          </svg>
        </span>
        tagB
      </a>
      <a
        className="tag is-paddingless"
        href="/blog/tags/tag-c/"
      >
        <span
          className="icon mr-0"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
            />
          </svg>
        </span>
        tagC
      </a>
    </div>
  </div>
</div>
`;

exports[`PostSummary renders post summary with excerpt 1`] = `
<div
  className="content"
>
  <h2
    className="subtitle is-marginless is-4"
  >
    <a
      href="/blogposttitle"
    >
      Post Title
    </a>
  </h2>
  <p
    className="is-marginless"
    dangerouslySetInnerHTML={
      Object {
        "__html": "The excerpt of the post.",
      }
    }
  />
  <div
    className="blog-metadata"
  >
    <div
      className="is-inline-block-tablet"
    >
      <small>
        <time
          dateTime="2020-05-08T12:58"
        >
          8 May 2020
        </time>
      </small>
    </div>
    <span
      className="blog-metadata-separator is-hidden-mobile"
    />
    <div
      className="tags is-inline-block-tablet"
    >
      <a
        className="tag is-paddingless"
        href="/blog/tags/tag-a/"
      >
        <span
          className="icon mr-0"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
            />
          </svg>
        </span>
        tagA
      </a>
      <a
        className="tag is-paddingless"
        href="/blog/tags/tag-b/"
      >
        <span
          className="icon mr-0"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
            />
          </svg>
        </span>
        tagB
      </a>
      <a
        className="tag is-paddingless"
        href="/blog/tags/tag-c/"
      >
        <span
          className="icon mr-0"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
            />
          </svg>
        </span>
        tagC
      </a>
    </div>
  </div>
</div>
`;
